lib/parser.rb

```diff
@@ -0,0 +1,32 @@
+# Understands how to parse natural language positions into formal ones
+
+class Parser
```

Comment:
Revealing intention: Watch out for generic words like parser.  Could this class name include a description of what it parses.